src/parser/monk/mistweaver/modules/talents/Lifecycles.tsx

```diff
@@ -61,11 +59,29 @@ class Lifecycles extends Analyzer {
 
   envelopingMistCast(event: CastEvent){
     // Checking to ensure player has cast Enveloping Mists and has the mana reduction buff
-    if (this.selectedCombatant.hasBuff(SPELLS.LIFECYCLES_ENVELOPING_MIST_BUFF.id)) {
-      this.manaSaved += SPELLS.ENVELOPING_MIST.manaCost * (SPELLS.LIFECYCLES_ENVELOPING_MIST_BUFF.manaPercRed);
-      this.manaSavedEnm += SPELLS.ENVELOPING_MIST.manaCost * (SPELLS.LIFECYCLES_ENVELOPING_MIST_BUFF.manaPercRed);
-      this.castsRedEnm += 1;
-      debug && console.log('ENM Reduced');
+    if(this.selectedCombatant.hasBuff(SPELLS.LIFECYCLES_ENVELOPING_MIST_BUFF.id)) {
+      if(this.selectedCombatant.hasBuff(SPELLS.INNERVATE.id)) {
+        return;
+      }
+      // Checking for chiji stacks and determine mana reduction
+      this.chijiStacksAtEnvCast = this.selectedCombatant.getBuff(SPELLS.INVOKE_CHIJI_THE_RED_CRANE_BUFF.id, event.timestamp)?.stacks;
+      if(this.chijiStacksAtEnvCast) {
+        //check for free cast from chiji
+        if(this.chijiStacksAtEnvCast === MAX_CHIJI_STACKS) {
+          return;
+        } else {
+          //have to do this weird because blizzard decided to make each chiji stack reduce the mana cost by 1001 instead of and exact 33%
```

Comment:
This is also wrong. In my testing they actually reduce it by 999 each stack which is 33.3%
Their system is probably doing 1/3 with a hard cap no rounding on 3 decimal places

Reply:
Where did you test? On beta it was reducing by 1001 consistently for me; 1 stack making it 1999 mana cost and 2 stacks making it 998